### PR TITLE
🐛  Fix "log.SetLogger(...) was never called" in registration webhook

### LINKED ISSUE #487: cluster-manager-registration-webhook logs can not be displayed
**Describe the bug**
After init a ocm hub, check the cluster-manager-registration-webhook logs, I got:
```
k logs -n open-cluster-management-hub cluster-manager-registration-webhook-56f69db8d8-drzlx
I0531 08:19:06.000786       1 start.go:77] "starting manager"
[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.
Detected at:
	>  goroutine 175 [running]:
	>  runtime/debug.Stack()
	>  	runtime/debug/stack.go:24 +0x5e
	>  sigs.k8s.io/controller-runtime/pkg/log.eventuallyFulfillRoot()
	>  	sigs.k8s.io/controller-runtime@v0.17.3/pkg/log/log.go:60 +0xcd
	>  sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).WithName(0xc00049e780, {0x2a1f5d0, 0x9})
	>  	sigs.k8s.io/controller-runtime@v0.17.3/pkg/log/deleg.go:147 +0x45
	>  github.com/go-logr/logr.Logger.WithName(...)
	>  	github.com/go-logr/logr@v1.4.1/logr.go:345
	>  sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).getLogger.func1()
	>  	sigs.k8s.io/controller-runtime@v0.17.3/pkg/webhook/admission/webhook.go:182 +0x66
	>  sync.(*Once).doSlow(0x0?, 0xc000532b90?)
	>  	sync/once.go:74 +0xbf
	>  sync.(*Once).Do(...)
	>  	sync/once.go:65
	>  sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).getLogger(0xc000a717c0, 0xc0002d3000?)
	>  	sigs.k8s.io/controller-runtime@v0.17.3/pkg/webhook/admission/webhook.go:180 +0x4d
	>  sigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).ServeHTTP(0xc000a717c0, {0x7f65ac2f9a38?, 0xc000936690}, 0xc00093a800)
	>  	sigs.k8s.io/controller-runtime@v0.17.3/pkg/webhook/admission/http.go:117 +0xe4b
	>  sigs.k8s.io/controller-runtime/pkg/webhook/internal/metrics.InstrumentedHook.InstrumentHandlerInFlight.func1({0x7f65ac2f9a38, 0xc000936690}, 0x2ecaa00?)
	>  	github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/instrument_server.go:60 +0xcb
	>  net/http.HandlerFunc.ServeHTTP(0x2ecaa10?, {0x7f65ac2f9a38?, 0xc000936690?}, 0xc000b838a0?)
	>  	net/http/server.go:2136 +0x29
	>  github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerCounter.func1({0x2ecaa10?, 0xc000942380?}, 0xc00093a800)
	>  	github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/instrument_server.go:147 +0xb6
	>  net/http.HandlerFunc.ServeHTTP(0x779086?, {0x2ecaa10?, 0xc000942380?}, 0xc00067c080?)
	>  	net/http/server.go:2136 +0x29
	>  github.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerDuration.func2({0x2ecaa10, 0xc000942380}, 0xc00093a800)
	>  	github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/instrument_server.go:109 +0xc2
	>  net/http.HandlerFunc.ServeHTTP(0x10?, {0x2ecaa10?, 0xc000942380?}, 0xc00067c3ac?)
	>  	net/http/server.go:2136 +0x29
	>  net/http.(*ServeMux).ServeHTTP(0x412885?, {0x2ecaa10, 0xc000942380}, 0xc00093a800)
	>  	net/http/server.go:2514 +0x142
	>  net/http.serverHandler.ServeHTTP({0x2ebb4e0?}, {0x2ecaa10?, 0xc000942380?}, 0x6?)
	>  	net/http/server.go:2938 +0x8e
	>  net/http.(*conn).serve(0xc00093c900, {0x2ee0030, 0xc00081c420})
	>  	net/http/server.go:2009 +0x5f4
	>  created by net/http.(*Server).Serve in goroutine 152
	>  	net/http/server.go:3086 +0x5cb
I0531 08:20:18.560342       1 log.go:245] http: TLS handshake error from 10.244.0.1:56545: EOF
```

**To Reproduce**

**Expected behavior**

**Environment ie: OCM version, Kubernetes version and provider:**

**Additional context**

### PR DESCRIPTION
**Pull request by RokibulHasan7**
## Related issue(s)

Fixes #487

### CODE REVIEW

File: pkg/registration/webhook/start.go [hunk 1]
```diff
@@ -45,6 +45,9 @@ func (c *Options) RunWebhookServer() error {
 	})
 	logger := klog.LoggerWithName(klog.FromContext(context.Background()), "Webhook Server") //MYTODO: Recheck it later
 
+	// This line prevents controller-runtime from complaining about log.SetLogger never being called
+	ctrl.SetLogger(logger)
+
```
Comment (zhujian7) [inline pkg/registration/webhook/start.go:50]: Do you think if it is better to move the logger init lines to the beginning of the func?
```suggestion
	logger := klog.LoggerWithName(klog.FromContext(context.Background()), "Webhook Server") //MYTODO: Recheck it later
	// This line prevents controller-runtime from complaining about log.SetLogger never being called
	ctrl.SetLogger(logger)

	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
		Scheme:                 scheme,
		HealthProbeBindAddress: ":8000",
		WebhookServer: webhook.NewServer(webhook.Options{
			Port:    c.Port,
			CertDir: c.CertDir,
			TLSOpts: []func(config *tls.Config){
				func(config *tls.Config) {
					config.MinVersion = tls.VersionTLS12
				},
			},
		}),
	})
	if err != nil {
		logger.Error(err, "unable to start manager")
		return err
	}
```

**Comment by zhujian7** [comment]
Thanks @RokibulHasan7 for the PR!!

**Comment by qiujian16** [comment]
/approve

Thanks!

**Pull request merged**